versioning_tools/lockstep_publish.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# Publish all subcrates to `crates.io` once we merged a version bump PR back to main.
```

Comment:
Version publishing can also be handled with tools like [release-plz](https://github.com/release-plz/release-plz) FYI

Reply:
Yeah it can, but the issue with `release-plz` is there's no way to do lockstep version bumping I think. From what I can tell (after reading through the config and playing with it locally), it will always bump each crate version individually based on semver changes for each crate.

The fact that I needed to write bash scripts in order to pull off automated lockstep versioning may be a warning sign I guess that maybe we shouldn't be doing lockstep to begin with. :sweat_smile: